wgpu-core/src/indirect_draw_validation.rs

```diff
@@ -0,0 +1,1080 @@
+use crate::{
+    device::{queue::TempResource, Device, DeviceError},
+    lock::{rank, Mutex},
+    pipeline::{CreateComputePipelineError, CreateShaderModuleError},
+    resource::{StagingBuffer, Trackable},
+    snatch::SnatchGuard,
+    track::TrackerIndex,
+    FastHashMap,
+};
+use std::{
+    mem::{size_of, size_of_val},
+    num::NonZeroU64,
+    sync::Arc,
+};
+use thiserror::Error;
+use wgt::Limits;
+
+/// Note: This needs to be under:
+///
+/// default max_compute_workgroups_per_dimension * size_of::<wgt::DrawIndirectArgs>() * `workgroup_size` used by the shader
+///
+/// = (2^16 - 1) * 2^4 * 2^6
+///
+/// It is currently set to:
+///
+/// = (2^16 - 1) * 2^4
+///
+/// This is enough space for:
+///
+/// - 65535 [`wgt::DrawIndirectArgs`] / [`MetadataEntry`]
+/// - 52428 [`wgt::DrawIndexedIndirectArgs`]
+const BUFFER_SIZE: wgt::BufferSize = unsafe { wgt::BufferSize::new_unchecked(1048560) };
```

Comment:
```suggestion
const BUFFER_SIZE: wgt::BufferSize = unsafe { wgt::BufferSize::new_unchecked(1_048_560) };
```

It'd be so nice to have this not be unsafe, but Option::unwrap is only const in 1.83